packages/plugins/yggdrasil-worker-forwarder/yggdrasil-worker-forwarder.spec

```diff
@@ -0,0 +1,57 @@
+%define debug_package %{nil}
+
+%if ! 0%{?gobuild:1}
+%define gobuild(o:) GO111MODULE=off go build -buildmode pie -compiler gc -tags="rpm_crashtraceback ${BUILDTAGS:-}" -ldflags "${LDFLAGS:-} -linkmode=external -B 0x$(head -c20 /dev/urandom|od -An -tx1|tr -d ' \\n') -extldflags '-Wl,-z,relro -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld '" -a -v %{?**};
+%endif
+
+Name: yggdrasil-worker-forwarder
+Version: 0.0.1
+Summary: Worker service for Yggdrasil that can forward requests to an API endpoint
+Release: 1%{?dist}
+License: GPLv3
+
+Source0: https://github.com/ShimShtein/%{name}/releases/download/v%{version}/%{name}-%{version}.tar.gz
+Url: https://github.com/%{repo_orgname}/%{name}/
+
+# EL7 doesn't define go_arches
+%if ! 0%{?go_arches:1}
+%define go_arches %{ix86} x86_64 %{arm} aarch64 ppc64le
+%endif
+ExclusiveArch: %{go_arches}
+
+BuildRequires: golang
+
+%description
+Worker service for Yggdrasil that can forward requests to an API endpoint
+
+%prep
+%autosetup
+
+%build
+%if 0%{?rhel} == 7
+mkdir -p _gopath/src/%{name}-%{version}
+cp -rf $(pwd)/*.go _gopath/src/%{name}-%{version}
+cp -rf $(pwd)/vendor _gopath/src/%{name}-%{version}
+export GOPATH=$(pwd)/_gopath
+pushd _gopath/src/%{name}-%{version}
+%{gobuild}
+strip %{name}-%{version}
+cp %{name}-%{version} %{_builddir}/%{name}-%{version}/%{name}
+popd
+%else
+%{gobuild}
+%endif
+
+%install
+mkdir -p %{buildroot}%{_bindir}
+%{__install} -m 755 %{_builddir}/%{name}-%{version}/%{name} %{buildroot}/%{_bindir}
+
+%files
+%{_bindir}/%{name}
```

Comment:
why `%_bindir`? I would have expected something like `/usr/libexec/yggdrasil`?

Reply:
In ygg/rhc 0.2.0+, a worker is no longer defined by an exec in the libexec directory but rather by a configuration file that points at the exec. That allows both re-use of the exec for multiple workers and putting the exec in a more natural location. Our goal is to plan ahead for this by installing it in a general location. Then it can either be symlinked or a worker file can be deployed to libexec that wraps and calls this workers code.

Reply:
Right, but those files are not user-executable (unless they prepare the environment to be exactly as yggd dictates), so I wouldn't expect them in `%_bindir`/`%_sbindir`.

Reply:
> Right, but those files are not user-executable (unless they prepare the environment to be exactly as yggd dictates), so I wouldn't expect them in `%_bindir`/`%_sbindir`.

You can pass all the right options to it and it work, and if yggd is running. Where would you expect them to live then? I was trying not to bank on libexec existing in the future *and* breakaway from having to figure out which piece of software we have.

Reply:
If we're to follow FHS, it should be either `/usr/libexec` or `/usr/lib/<tool>`.

I won't block on this, but bin seems wrong.

Reply:
Aight - let me know what you think about this version.

Reply:
I like that :)